src/node.ts

```diff
@@ -0,0 +1,396 @@
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'GetCode', args.address, cb);
+  };
+
+  createMessage = (args, cb) => {
+    try {
+      validateArgs(
+        {
+          to: [util.isAddress],
+        },
+        {
+          from: [util.isAddress],
+          gas: [util.isNumber],
+          gasPrice: [util.isNumber],
+        },
+      );
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'CreateMessage', args, cb);
+  };
+
+  getGasEstimate = (args, cb) => {
+    try {
+      validateArgs(
+        {},
+        {
+          to: [util.isAddress],
+          from: [util.isAddress],
+          gas: [util.isNumber],
+          gasPrice: [util.isNumber],
+          gasLimit: [util.isNumber],
+        },
+      );
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'GetGasEstimate', args, cb);
+  };
+
+  getTransactionReceipt = (args, cb) => {
+    try {
+      validateArgs(args, {
+        txHash: [util.isHash],
+      });
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'GetTransactionReceipt', args.txHash, cb);
+  };
+
+  getHashrate = (cb) => {
+    rpcAjax(this.url, 'GetHashrate', '', cb);
+  };
+
+  isNodeMining = (args, cb) => {
+    rpcAjax(this.url, 'isNodeMining', '', cb);
+  };
+
+  compileCode = (args, cb) => {
+    try {
+      validateArgs(args, {
+        code: [util.isString],
+      });
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'CompileCode', args, cb);
+  };
+
+  checkCode = (args, cb) => {
+    try {
+      validateArgs(args, {
+        code: [util.isString],
+      });
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    serverAjax(this.apiUrl + '/v1/checker', args, cb);
+  };
+
+  checkCodeTest = (args, cb) => {
+    try {
+      validateArgs(args, {
+        code: [util.isString],
+      });
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    serverAjax(this.apiUrl + '/v1/runner', args, cb);
+  };
+
+  // Explorer APIs
+  getBlockchainInfo = (args, cb) => {
+    rpcAjax(this.url, 'GetBlockchainInfo', '', cb);
+  };
+
+  getDSBlockListing = (args, cb) => {
+    try {
+      validateArgs(args, {
+        page: [util.isNumber],
+      });
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'DSBlockListing', args.page, cb);
+  };
+
+  getTxBlockListing = (args, cb) => {
+    try {
+      validateArgs(args, {
+        page: [util.isNumber],
+      });
+    } catch (e) {
+      cb(e);
+      return;
+    }
+
+    rpcAjax(this.url, 'TxBlockListing', args.page, cb);
+  };
+
+  getNumTxnsTxEpoch = (args, cb) => {
+    rpcAjax(this.url, 'GetNumTxnsTxEpoch', '', cb);
+  };
+
+  getNumTxnsDSEpoch = (args, cb) => {
+    rpcAjax(this.url, 'GetNumTxnsDSEpoch', '', cb);
+  };
+
+  getTransactionListing = (args, cb) => {
+    rpcAjax(this.url, 'GetRecentTransactions', '', cb);
+  };
+}
+
+function rpcAjax(url, method, params, cb) {
+  postData(url, {
```

Comment:
since you are using a promise approach, it would be nice to return the promise and make `cb` optional. This way  the function could be used as promise or with callback.

I'm thinking of something like this:
```
function rpcAjax(url, method, params, cb?): Promise<any> {
    const promise = postData(...);

    if (cb && typeof cb === "function") {
        promise
            .then(data => cb(null, data))
            .catch(error => cb(error));
    }

    return promise;
}
```

Reply:
For now, this is to avoid breaking downstream dependents.

Reply:
But you don't brake anything, and with my proposal you can do a smooth transition.